lib/api.js

```diff
@@ -8,100 +8,93 @@ var sec = require('fh-security'),
 var url = require('url');
 var packageJSON = require('../package.json');
 var initScript = require('./init.js');
-
+var EventEmitter = require("events");
 var mbaasClient = require('fh-mbaas-client');
 
 //IMPORTANT: This will force node to ignore cert errors for https requests
 process.env.NODE_TLS_REJECT_UNAUTHORIZED = "0";
 
-//
-// Main FHapi constructor function..
-//
-function FHapi(cfg, logr) {
+function createAPI(cfg, logger) {
   if (cfg) {
-    cfg.logger = logr;
+    cfg.logger = logger;
   }
 
-  var api = {
-    getVersion: function(){
-      //Getting The Version of fh-mbaas-api
-      return packageJSON.version;
-    },
-    cache: require('./cache')(cfg),
-    db: require('./db')(cfg),
-    forms: require('./forms')(cfg),
-    log: false,
-    stringify: false,
-    parse: false,
-    push: require('./push')(cfg),
-    call: require('./call')(cfg),
-    util: false,
-    redisPort: cfg.redis.port || '6379',
-    redisHost: cfg.redis.host || 'localhost',
-    session: require('./session')(cfg),
-    stats: require('./stats')(cfg),
-    sync: require('./sync').api,
-    act: require('./act')(cfg),
-    service: require('./act')(cfg),
-    sec: sec.security,
-    auth: require('./auth')(cfg),
-    host: require('./host'),
-    permission_map: require('fh-db').permission_map,
-    hash: function (opts, callback) {
-      var p = {
-        act: 'hash',
-        params: opts
-      };
-      sec.security(p, callback);
-    },
-    web: require('./web')(cfg)
+  //
+  // Main FHapi constructor function..
+  //
+  function FHapi() {
```

Comment:
@evanshortiss That change is from https://github.com/feedhenry/fh-mbaas-api/pull/101. It is using prototype so the api object being returned to the developer can extend EventEmitter.
However, this may be superceded by your PR #80 where `sync.events` is an EventEmitter